tarantool_test.go

```diff
@@ -4019,6 +4019,38 @@ func TestConnect_schema_update(t *testing.T) {
 	}
 }
 
+func TestConnect_context_cancel(t *testing.T) {
+	var connLongReconnectOpts = Opts{
+		Timeout:       5 * time.Second,
+		User:          "test",
+		Pass:          "test",
+		Reconnect:     time.Second,
+		MaxReconnects: 100,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var conn *Connection
+	var err error
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		cancel()
+		conn, err = Connect(ctx, server, connLongReconnectOpts)
+	}()
+
+	cancel()
```

Comment:
Why do you call cancel() twice? Why we need the goroutine?

Reply:
Oops, updated.